rust/lance-index/src/scalar/json.rs

```diff
@@ -779,40 +784,72 @@ impl ScalarIndexPlugin for JsonIndexPlugin {
             .unwrap();
         let path = request.parameters.path.clone();
 
-        // Extract JSON with type information
-        let (data_stream, inferred_type) =
-            Self::extract_json_with_type_info(data, path.clone()).await?;
-
-        // Convert the stream to properly typed values based on inferred type
-        let converted_stream =
-            Self::convert_stream_by_type(data_stream, inferred_type.clone()).await?;
-
-        // Update the target request with inferred type
-        let registry = self.registry()?;
-        let target_plugin = registry.get_plugin_by_name(&request.parameters.target_index_type)?;
-
-        // Create a new training request with the inferred type
-        let target_request = target_plugin.new_training_request(
-            request
-                .parameters
-                .target_index_parameters
-                .as_deref()
-                .unwrap_or("{}"),
-            &Field::new("", inferred_type, true),
-        )?;
-
-        let target_index = target_plugin
-            .train_index(converted_stream, index_store, target_request, fragment_ids)
-            .await?;
-
-        let index_details = crate::pb::JsonIndexDetails {
-            path,
-            target_details: Some(target_index.index_details),
-        };
-        Ok(CreatedIndex {
-            index_details: prost_types::Any::from_msg(&index_details)?,
-            index_version: JSON_INDEX_VERSION,
-        })
+        match request.parameters.target_index_type.as_str() {
```

Comment:
Can you make a new plugin for this instead of sharing the plugin?  These are two rather different index types.